source/components/molecules/ImageUploader/ImageUploader.tsx

```diff
@@ -118,13 +119,25 @@ const ImageUploader: React.FC<Props> = ({
   colorSchema,
   maxImages,
   id,
+  imageNamePrefix,
+  ...rest
 }) => {
   const [choiceModalVisible, toggleModal] = useModal();
 
+  const renameImagesWithPrefix = (images: Image[], prefix: string) =>
+    images.map((image, idx) => ({
+      ...image,
+      filename: `${prefix}_${idx}`,
+    }));
+
   const addImagesToState = (newImages: Image[]) => {
-    const updatedImages =
+    let updatedImages =
       images === "" ? [...newImages] : [...images, ...newImages];
 
+    if (imageNamePrefix) {
+      updatedImages = renameImagesWithPrefix(updatedImages, imageNamePrefix);
+    }
+
```

Comment:
**OPTIONAL**

One (optional) thing you could do is to give `imageNamePrefix` a default value (imageNamePrefix = ''), then you do not need to have the if-statement for checking if the value is set.

Reply:
If I don't have the check, and instead only use the `renameImagesWithPrefix` function, the original filenames will not be preserved when the `imageNamePrefix` is not set, but will instead be named in the pattern `_${idx}`.